notebooks/region_search/Region Searching Workbook.ipynb

```diff
@@ -1754,6 +1755,108 @@
     "df[\"ut\"]"
    ]
   },
+  {
+   "cell_type": "code",
+   "execution_count": 129,
+   "id": "b45d3fa2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# As a reality check, let's make sure there aren't two identical dataId + UT instances\n",
+    "def look_for_duplicates(df):\n",
+    "    \"\"\"3/11/2024 COC\"\"\"\n",
+    "    # make sure no duplicate URIs first (easy + fast)\n",
+    "    if len(df[\"uri\"]) != len(set(df[\"uri\"])):\n",
+    "        assert AttributeError(\n",
+    "            f'There were only {len(set(df[\"uri\"]))} unique URIs. Should have been {len(df[\"uri\"])}.'\n",
+    "        )\n",
+    "    #\n",
+    "    if len(df[\"data_id\"]) != len(set(df[\"data_id\"])):\n",
+    "        assert AttributeError(f\"Encountered non-unique set of dataIds.\")\n",
+    "    #\n",
+    "    dataId_uts = df[\"ut\"].astype(str) + df[\"data_id\"].astype(\n",
+    "        str\n",
+    "    )  # e.g., \"2019-09-27T00:20:22.932{instrument: 'DECam', detector: 1, visit: 898286}\"\n",
+    "    if len(dataId_uts) != len(set(dataId_uts)):\n",
+    "        assert AttributeError(f\"dataID + UT combinations were not unique\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 132,
+   "id": "c3beca0f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "assert len(df[\"data_id\"]) == len(set(df[\"data_id\"]))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 122,
+   "id": "381e68de",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "Index(['data_id', 'region', 'detector', 'uri', 'ut', 'center_coord',\n",
+       "       'ut_datetime'],\n",
+       "      dtype='object')"
+      ]
```

Comment:
I would remove these cells (screenshot attached) in favor of a single call to your duplicate-checking function, but understand you may want it this way to demonstrate exploration of the dataframe.
 
<img width="977" alt="Screenshot 2024-03-13 at 5 08 54 PM" src="https://github.com/dirac-institute/kbmod/assets/22227694/9e17d4cc-0270-4aef-a4e6-22f3b44b1ca9">